tony-history-server/app/utils/HdfsUtils.java

```diff
@@ -63,27 +59,52 @@ static String contentOfHdfsFile(FileSystem fs, Path filePath) {
   }
 
   /**
-   * List all metadata file paths in {@code histFolder}.
+   * Extract the job id portion from the file path string.
+   * @param path file path string.
+   * @return job id
+   */
+  public static String getJobId(String path) {
+    if (path.length() == 0) return "";
+    String[] folderLayers = path.split("/");
+    return folderLayers[folderLayers.length - 1];
+  }
+
+  /**
+   * Check to see if the path {@code p} is a job folder path from the given
+   * folder name pattern {@code regex}
+   * @param p Path object
+   * @param regex regular expression string
+   * @return true if path {@code p} is a job folder path, false otherwise.
+   */
+  @VisibleForTesting
+  static boolean isJobFolder(Path p, String regex) {
+    return getJobId(p.toString()).matches(regex);
+  }
+
+  /**
+   * Find all job folders under {@code curr} that matches {@code regex} pattern
```

Comment:
It'd be helpful to mention that this method is recursive

Reply:
Sure 👍

Reply:
Fixed: https://github.com/linkedin/TonY/pull/81/commits/108fe8650f76b5d9d5ae71188f3b95d170b54a93.